curriculum_algorithm/admin.py

```diff
@@ -0,0 +1,32 @@
+from django.contrib import admin
+
+from curriculum_algorithm.models import Curriculum, Course, CurriculumCourse, StudentPlan, Semester
+
+
+class CurriculumCourseInline(admin.TabularInline):
+    fields = ['course', 'pre_requisites', 'co_requisites']
+    fk_name = 'curriculum'
+    model = CurriculumCourse
+
+
+@admin.register(Curriculum)
+class CurriculumAdmin(admin.ModelAdmin):
+    fields = ['name', 'department']
+    inlines = [CurriculumCourseInline]
+
+
+class SemesterInline(admin.TabularInline):
+    fields = ['curriculum_courses', 'credit_hours', 'position',  'past']
+    fk_name = 'student_plan'
+    model = Semester
+    extra = 0
+
+
+@admin.register(StudentPlan)
+class StudentPlanAdmin(admin.ModelAdmin):
+    fields = ['max_credits', 'curriculum']
+    inlines = [SemesterInline]
+
```

Comment:
In the future it would be cool if in the admin change list of Student Plans you could see number of credits and curriculum name. Maybe adding list_display = ('curriculum', 'max_credits') works

Reply:
The admin page is just a placeholder. The final  design will most likely be a RESTful interface and the front-end will be managed as a separate project.